Starting to update roadmap document

nickerso opened the pull request:
Starting work on updating the roadmap document. Not yet ready to merge, but maybe useful for @hsorby and @agarny to take a look and feel free to contribute ;-)

hsorby committed:
Start to add the notion of equality to entities.

nickerso committed:
Merge branch 'develop' into equality

hsorby committed:
Merge in changes from 'prime/develop' fixing conflicts.

hsorby committed:
Progress towards adding equal function to all entities.

hsorby committed:
Adding equal logic to more entities.

hsorby committed:
Refuse to add the same entity repeatedly.

hsorby committed:
Revert changes to the private implementation of entities.

hsorby committed:
Merge branch 'equality' of github.com:hsorby/libcellml into equality

hsorby committed:
Add test for variables differing in units.

hsorby committed:
Add another test for variables with differeing units.

hsorby committed:
Add const where appropriate to find equal methods.

hsorby committed:
Cast unsigned long to signed long in vector offsetting.

hsorby committed:
Replace most int64_t with ssize_t (except sha1 calculation ones).

hsorby committed:
Address clang-tidy error.

hsorby committed:
Add a basic double comparison function.  Add note to add* methods on maximum number of additions assumed.

hsorby committed:
Consider more advanced nearlyEqual function.

hsorby committed:
Fix failing tests.

hsorby committed:
Correct clang tidy error.

hsorby committed:
Fix compilation errors.

hsorby committed:
Apply clang formatting to codebase.

hsorby committed:
Add Python test for equal.

hsorby committed:
Add doxygen documentation to doEqual and equal methods.

hsorby committed:
Remove whitespace from end of line in variable.h.

hsorby committed:
Fix up private doxygen documentation to be consistent.

hsorby committed:
Add test for equality of slightly modified HH model.

hsorby committed:
Minor comments added for add* methods.

hsorby committed:
Update entity.h

hsorby committed:
Revert changes to tests that shouldn't have been changed.

hsorby committed:
Merge branch 'equality' of github.com:hsorby/libcellml into equality

hsorby committed:
Address review comments. Add reverse equality test for completness.

hsorby committed:
Fixes to documentation and protection against adding very large numbers of entities.

hsorby committed:
Cast to compare unsinged integers and not a mix.

hsorby committed:
Rationalise the math equality into a function.

hsorby committed:
Rename internal math.{h,cpp} to mathml.{h,cpp}.

hsorby committed:
Fix typo in compareMath doxygen documentation.

hsorby committed:
Merge branch 'develop' into equality

hsorby committed:
Change 'meter' for 'metre'.

hsorby committed:
Rename 'equal' to 'equals'.

hsorby committed:
Remove duplicate test variableNotEqualByUnitsOther.

hsorby committed:
Start to add test for making sure parsing a model doesn't prettify the line endings.

hsorby committed:
Update python tests to use 'equals' method.

hsorby committed:
Add test to show math is treated as a straight string.

hsorby committed:
Remove MathML cleaning functions from parser.

hsorby committed:
Merge branch 'equality' of github.com:hsorby/libcellml into equality

hsorby committed:
Add test for flattening.

hsorby committed:
Remove unused code from equality tests.

hsorby committed:
Remove note on limit to the number of entities can be added.

hsorby committed:
Address review comments for entity.h.

hsorby committed:
Remove class ImportedEntityPrivate;.

hsorby committed:
Address readability concerns.

hsorby committed:
Refactor equalUnits, equalVariables, equalResets.

hsorby committed:
Replace doEqual with doEquals.

hsorby committed:
Remove mathml.{h.cpp} and change compareMath to areEqual.

hsorby committed:
Tidy doEquals method in units.cpp.

hsorby committed:
Simplify equality tests for resets.

hsorby committed:
xor is not a cross-platform operator.  Replace with equivalent.

hsorby committed:
No else if after return in reset.cpp.

hsorby committed:
Fix compilation errors on macOS.

hsorby committed:
Fix clang formatting errors.

hsorby committed:
Add annotation and correct typos in equalEntities documentation.

hsorby committed:
Rename 'nearlyEqual' to 'areNearlyEqual'.

hsorby committed:
Remove ! from in front of isinf.

hsorby committed:
Remove the restriction about adding the same pointer twice.

hsorby committed:
Correct documentation for add* methods.

nickerso committed:
add missing header needed for windows compile

nickerso committed:
Add a couple of tests demonstrating multiple add behaviour.

With the change in behaviour introduced with the ability to add entities multiple times
it would seem useful to test for expected behaviour in case users come to depend on this
funky use case continuing to function consistently.

hsorby committed:
Merge pull request #12 from nickerso/equality

Windows compile and new tests

nickerso committed:
Starting to update roadmap document

hsorby committed:
Update roadmap.rst

nickerso committed:
Merge pull request #6 from hsorby/patch-1

Update roadmap.rst

hsorby commented:
It looks pretty good to me, and still relevant (not sure if that is a good or bad thing). Does it need more or is the scope still about right?

hsorby committed:
Apply clang format to src/variable.cpp

agarny committed:
Utilities: reworked areEqual() and replaced some areEqual() calls with areNearlyEqual() calls.

agarny committed:
Utilities: reworked areEqual() and replaced some areEqual() calls with areNearlyEqual() calls.

Make Linux happy.

hsorby committed:
Merge pull request #785 from hsorby/equality

Add the notion of equality to entities.  Fixes #719.

nickerso committed:
Merge branch 'main' into roadmap-update

nickerso committed:
Merge branch 'roadmap-update' of https://github.com/nickerso/libcellml into roadmap-update

nickerso committed:
Update README with minor tweaks

Updating README to now point to libcellml.org rather than readthedocs and to bring things into line with the current status.

nickerso committed:
fixing broken buildbot links
